components/Operations/tezConnections.ts

```diff
@@ -113,11 +129,29 @@ export function updateStakingOpsStatus(
       return operation
     })
   }
-  // if prevBakerAddress is not null then check if the current baker address matches the prevBakerAddress then only CanStake should be true.
-  if (prevBakerAddress) {
-    opStatus.CanStake =
-      opStatus.CanStake && prevBakerAddress === delegate.address
+
+  if (type === 'user') {
+    opStatus.pendingUnstakeOpsWithAnotherBaker =
+      (prevBakerAddress &&
+        (!Boolean(delegate) || prevBakerAddress !== delegate?.address)) ??
+      false
+    opStatus.bakerAcceptsStaking =
+      (Boolean(delegate) &&
+        bakerList?.find(baker => baker.address === delegate.address)
+          ?.acceptsStaking) ??
+      false
   }
 
+  if (type === 'delegate') {
+    opStatus.pendingUnstakeOpsWithAnotherBaker =
+      (prevBakerAddress && prevBakerAddress !== accountInfo.address) ?? false
```

Comment:
can simplfy to `prevBakerAddress && prevBakerAddress !== accountInfo.address`

Reply:
It throws an error. But following worked
` Boolean(prevBakerAddress) && prevBakerAddress !== accountInfo.address`